php/libraries/Utility.class.inc

```diff
@@ -1001,20 +1001,21 @@ class Utility
     /**
      * Get the list of MRI scan type available in the database
      *
+     * @param bool $join_files Whether to perform an inner join on the files table
+     *
      * @return array Array of MRI scan types which exist in the database table
      *               mri_scan_type joined to the files table
      *               array($scan_type_id => $scan_type)
      */
-    static function getScanTypeList(): array
+    static function getScanTypeList($join_files = true): array
     {
         $DB = \Database::singleton();
-
-        $scan_types_DB = $DB->pselect(
-            "SELECT ID, Scan_type 
-             FROM mri_scan_type mri
-                JOIN files f ON (f.AcquisitionProtocolID=mri.ID)",
-            array()
-        );
+        $query = "SELECT ID, Scan_type
+             FROM mri_scan_type mri\n";
+        if ($join_files) {
+            $query .= "JOIN files f ON (f.AcquisitionProtocolID=mri.ID)";
```

Comment:
Karim can you explain in your PR description what you're solving, why you're solving it like this, and what this query changes?   Like -- adding a parameter to join with the files table to return... (and _why_) 
This is important to state clearly upfront as there are potential side effects across the codebase where this function is called.

Reply:
Sure thing, I added more details to the PR description, let me know if there's anything more missing.